src/quacc/recipes/vasp/_base.py

```diff
@@ -120,3 +123,74 @@ def run_and_summarize_opt(
         report_mp_corrections=report_mp_corrections,
         additional_fields=additional_fields,
     )
+
+
+def run_and_summarize_vib_and_thermo(
+    atoms: Atoms,
+    energy: float = 0.0,
+    temperature: float = 298.15,
+    pressure: float = 1.0,
+    thermo_method: Literal["ideal_gas", "harmonic"] = "ideal_gas",
+    preset: str | None = None,
+    calc_defaults: dict[str, Any] | None = None,
+    calc_swaps: dict[str, Any] | None = None,
+    vib_kwargs: dict[str, Any] | None = None,
+    copy_files: SourceDirectory | dict[SourceDirectory, Filenames] | None = None,
+    additional_fields: dict[str, Any] | None = None,
+) -> VibThermoSchema:
+    """
+    Base job function for VASP recipes with ASE vibrational analysis.
+
+    Parameters
+    ----------
+    atoms
+        Atoms object
+    energy
+        Energy of the system
+    temperature
+        Temperature of the system
+    pressure
+        Pressure of the system
+    preset
+        Preset to use from `quacc.calculators.vasp.presets`.
+    calc_defaults
+        Default parameters for the recipe.
+    calc_swaps
+        Dictionary of custom kwargs for the Vasp calculator. Set a value to
+        `None` to remove a pre-existing key entirely. For a list of available
+        keys, refer to [quacc.calculators.vasp.vasp.Vasp][].
+    vib_kwargs
+        Dictionary of custom kwargs for [quacc.runners.ase.Runner.run_vib][]
+    copy_files
+        Files to copy (and decompress) from source to the runtime directory.
+
+    Returns
+    -------
+    VibThermoSchema
+        Dictionary of results, specified in [quacc.schemas.ase.summarize_vib_and_thermo][].
+        See the type-hint for the data structure.
+    """
+
+    # Set defaults
+    calc_flags = recursive_dict_merge(calc_defaults, calc_swaps)
+
+    calc = Vasp(atoms, preset=preset, **calc_flags)
+    vibrations = Runner(atoms, calc, copy_files=copy_files).run_vib(
+        vib_kwargs=vib_kwargs
+    )
+    if thermo_method == "harmonic":
+        thermo_analysis = ThermoRunner(
+            atoms, vibrations.get_frequencies(), energy=energy
+        ).run_harmonic()
```

Comment:
Could you also add one very small test to run a VASP recipe with the harmonic method so this gets covered? You have already tested the harmonic thermo functionality elsewhere, so this can be pretty minimal --- just to make sure we don't break it later.